[HPR-817] command/validate: add option to eval datasources (#12106)
hashicorp/packer: website/content/docs/commands/validate.mdx

- `-syntax-only` - Only the syntax of the template is checked. The
  configuration is not validated.

> - `-evaluate-datasources` - Evaluate the datasources when validating the template.
>   This is only valid on HCL2 templates, since JSON templates do not feature
>   datasources, this option will be ignored. Note that because datasources may rely
>   on external services for fetching data, this will incur some costs at validation
>   when the services being contacted are billing for the operation.
> 

Reviewer: Slight reword with a call out for the billing warning 
```suggestion
- `-evaluate-datasources` - Evaluate all data sources when validating a template.
  This is only valid on HCL2 templates, since JSON templates do not feature
  data sources. For JSON templates this flag will be ignored. 
  
  ~> **Warning:** Data sources may rely on external services for fetching data, which can incur some costs at validation if the services being contacted are billing per operation.
```

  source block's "name" label, unless an in-build source definition adds the
  "name" configuration option.

> - `-no-warn-on-undeclared-var` - Silence warnings when the template assigns
>   undeclared variables. This is especially relevant for HCL2 templates with
>   .pkrvars.hcl dependencies, as defining a value for a variable is not enough on
>   its own for Packer to function, there also needs to be some variable definitions
>   in the template. However, for some cases, templates may only declare what they
>   locally need for building, and reference a global variable file with more variable
>   assignations than there are definitions. This would yield a warning for each
>   assigned and non-declared variable. This option will disable those warnings.
> 

Reviewer: ```suggestion
- `-no-warn-on-undeclared-var` - Silence warnings when the a variable definition file contains variable assignments for undeclared variables. This can occur when using a var-file that contains a large amount of unused variables for a given HCL2 template. For HCL2 template defining a value for a variable in a var-file is not enough on its own for Packer to function, as there also needs to be a variable block definition in the template files `pkr.hcl` for the variable. By default `packer validate` will warn when a var-file contains one or more undeclared variables. 
```
